ethtool: use ioctl wrapper from golang.org/x/sys/unix

tklauser opened the pull request:
Update the vendored copy of `golang.org/x/sys/unix` to pull inioctl wrapper fixes and switch `pkg/datapath/ethtool` to use one of these wrappers with the correctly padded `ifreqData struct`.

See individual commits for details.

ti-mo commented on pkg/datapath/linux/ethtool/ethtool_linux.go:
```diff
@@ -0,0 +1,24 @@
+	if err != nil {
+		return false, err
+	}
+	return string(bytes.TrimRight(info.Driver[:], "\x00")) == "veth", nil
```
Is this only ever called on veth ifs? Shouldn't this include bridges and mac/ipvlan as well? If not, the fn name might be a bit misleading.

```
# ethtool -i eth1
driver: macvlan
...
```

tklauser replied:
Currently, the only call site is here:

https://github.com/cilium/cilium/blob/2d075ea34720ed625e297ede67ace8dce5bbdb53/pkg/datapath/loader/base.go#L182

used in ENI IPAM mode top determine which interfaces to use for IPSec encryption. I'm not very familiar with that, so I'd have to defer to @jrfastab to answer the question of whether we should include bridges and mac/ipvlan as well.

tklauser committed:
vendor: bump golang.org/x/sys to latest version

This pulls in a few fixes around ioctl wrappers wrt. unsafe.Pointer
usage and fixes ifreqEthtool to be correctly padded.

Ref. golang/sys@e5e7981a10699f0af2ffea4c0e0f542e447b2e4a
Ref. golang/sys@b4502255bfe74b77ffdeb6ed972eeecd98d13721

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
datapath/linux/ethtool: use IoctlGetEthtoolDrvinfo

Use the ioctl wrapper provided in the golang.org/x/sys/unix package with
the correctly padded ifreqData struct, rather than providing our own
wrapper and struct which is incorrectly padded.

Also add a simple unit test and make sure the package is only built on
Linux.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

kaworu committed:
Merge branch 'master' into ethtool-x-sys-unix
